client/components/common/SideNavDrawer.tsx

```diff
@@ -31,21 +29,22 @@ export default function SideNavDrawer({
     animation: 'opacity-out 200ms ease-in',
     animationFillMode: 'forwards',
   };
-  const { data: userProfile, isError, error } = useFetchUserProfile();
-  const { showNotification } = useNotification();
 
-  if (isError) {
-    console.error('Failed to fetch user profile: ', error);
-    let errorMessage = 'Failed to fetch user profile. Please try again.';
-
-    if (error instanceof AxiosError && error.response) {
-      errorMessage = error.response.data || errorMessage;
-    } else if (error instanceof Error) {
-      errorMessage = error.message;
+  const { data: userProfile, status } = useFetchUserProfile();
+
+  const renderUserProfile = () => {
+    if (status === 'pending') {
+      return <Typography variant='caption'>Loading...</Typography>;
+    } else if (
+      status === 'success' &&
+      typeof userProfile !== 'string' &&
+      userProfile?.name
+    ) {
+      return <Typography variant='caption'>{userProfile.name}</Typography>;
+    } else {
+      return <Typography variant='caption'>Guest</Typography>;
```

Comment:
I’m not sure I fully understand the reason behind using `guest` here.  Is this a placeholder until we get a fix for user creation bug?  I think it could be problematic down the road to have users autenticated as a guest to use our app but not exisiting our database.

Reply:
Thanks for calling this out. I think this implementation can be improved for better UX. It is a bit confusing to render Guest for error state. I will make some changes to make it more intuitive.